go/vt/vtgate/executor_select_test.go

```diff
@@ -508,12 +508,9 @@ func TestLastInsertIDInSubQueryExpression(t *testing.T) {
 	sbc1.SetResults(result1)
 	_, err := executorExec(executor, "select (select last_insert_id()) as x", nil)
 	require.NoError(t, err)
-	wantQueries := []*querypb.BoundQuery{{
-		Sql:           "select (select :__lastInsertId as `last_insert_id()` from dual) as x from dual",
-		BindVariables: map[string]*querypb.BindVariable{"__lastInsertId": sqltypes.Uint64BindVariable(0)},
-	}}
 
-	assert.Equal(t, wantQueries, sbc1.Queries)
+	// the query will get rewritten into a simpler query that can be run entirely on the vtgate
+	assert.Empty(t, sbc1.Queries)
 }
```

Comment:
1. This test does not require sbc1.SetResults as it does not go to sandbox connections.
1. Validate the query result of executed query it should match what is there in the session lastInsertId